app/src/main/java/com/sdp/movemeet/HomeScreenActivity.java

```diff
@@ -12,23 +17,37 @@
 import com.sdp.movemeet.map.GPSRecordingActivity;
 
 import com.sdp.movemeet.utility.ActivitiesUpdater;
+import com.sdp.movemeet.utility.LocationFetcher;
+
 
 public class HomeScreenActivity extends AppCompatActivity {
 
     private FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
 
+    private ActivityResultLauncher<String> requestLocationPermissionLauncher = registerForActivityResult(new ActivityResultContracts.RequestPermission(), isGranted -> {
+        if (!isGranted) {
+            LocationFetcher.currentLocation = LocationFetcher.defaultLocation();
+        }
+    });
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_home_screen);
 
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            requestLocationPermissionLauncher.launch(Manifest.permission.ACCESS_FINE_LOCATION);
+        }
+
         ActivitiesUpdater updater = ActivitiesUpdater.getInstance();
-        updater.updateListActivities();
+        // Always clear activities first, to prevent duplicates if multiple intents are created
```

Comment:
👍

Reply:
yes, I noticed that when the user came back to the homescreen after being logged out, it made a new intent but kept the old list as it was. So this created duplicates.